tools/webots.py

```diff
@@ -98,11 +98,8 @@ def register(command):
 
 def get_cmake_flags(roles_to_build):
 
-    available_roles = []
-    for dirpath, _, fnames in os.walk("roles"):  # iterate over the roles directory and all subdirectories
-        for f in fnames:  # iterate over all the files in the directory
-            if f.endswith(".role"):  # check that the file is a role file
-                available_roles.append(os.path.join(dirpath, f))  # add the role file to our list
+    # Find all available roles
+    available_roles = available_roles = glob(os.path.join("roles", "**", "*.role"), recursive=True)
```

Comment:
```suggestion
    available_roles = glob(os.path.join("roles", "**", "*.role"), recursive=True)
```
:stuck_out_tongue:

Also do we need to strip the `roles/` and `.role` from the filename https://github.com/NUbots/NUbots/pull/1010#discussion_r1068685321?

Reply:
Don't assume that the b script is always running from the project root directory, while that is commonly the case it doesn't have to be.

`available_roles = glob(os.path.join(b.project_dir, "roles", "**", "*.role"), recursive=True)`

Reply:
`available_roles = available_roles = glob(os.path.join("roles", "**", "*.role"), recursive=True)`
I already fixed this.

Reply:
> Don't assume that the b script is always running from the project root directory, while that is commonly the case it doesn't have to be.
> 
> `available_roles = glob(os.path.join(b.project_dir, "roles", "**", "*.role"), recursive=True)`

I'll add this when i can test it

Reply:
'twerks